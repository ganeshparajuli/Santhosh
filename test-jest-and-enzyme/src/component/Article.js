import React, { Component } from 'react'
import Table from './Table'
export class Article extends Component {
    constructor(props){
        super(props)
        this.state={
            count: 0
        }
    }
    render() {
        const {count}=this.state;
        return (
            <div>
                <div className="clicks-1"></div>
                <div className={`clicks-${count}`}>{count}</div>
                <button onClick={()=>{this.setState({count: count + 1 }); }}>Increment</button>
                <h1>Article Details</h1>
                <Table />
            </div>
        )
    }
}

export default Article
