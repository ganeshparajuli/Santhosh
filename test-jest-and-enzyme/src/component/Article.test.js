import Article from "./Article";
import { shallow } from "enzyme";
import Table from "./Table"

describe('check Article page rendered.',()=>{
let wrapper=shallow(<Article />)

test("if click count proper.",()=>{
    expect(wrapper.find('.clicks-0').length).toEqual(1);
    wrapper.find('button').simulate('click');
    expect(wrapper.find('.clicks-1').length).toEqual(2);
})

test("using dive.",()=>{
    expect(wrapper.find(Table).dive().find('.custom-border').length).toBe(1);  
})
test("using snapshot 1.",()=>{
    let wrapper=shallow(<Table />)
    expect(wrapper).toMatchSnapshot();
})

})