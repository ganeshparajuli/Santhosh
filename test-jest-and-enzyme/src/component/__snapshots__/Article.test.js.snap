// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check Article page rendered. using snapshot 1. 1`] = `
<table
  className="custom-border"
>
  <thead>
    <tr>
      <th>
        Name
      </th>
      <th>
        Email
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        Demo
      </td>
      <td>
        test@demo.com
      </td>
    </tr>
  </tbody>
</table>
`;
